counter/src/main/java/org/infinispan/counter/impl/manager/CounterConfigurationManager.java

```diff
@@ -236,10 +236,10 @@ private void startCounterCache() {
                .execute(() -> {
                   String oldName = Thread.currentThread().getName();
                   try {
-                     Thread.currentThread().setName(threadName());
+                     SecurityActions.setThreadName(threadName());
```

Comment:
I believe the original issue was that in `threadName` we get the cache configuration, which also requires a security action.

Reply:
That makes more sense :see_no_evil:

Reply:
Maybe keep this too? It's probably pointless because we have so many places where we create and rename threads without privileged actions, but still :)

Reply:
It's already removed now and like you say there are probably many other places.